engine/crates/grafbase-engine/src/resolver_utils/field.rs

```diff
@@ -50,13 +50,24 @@ pub async fn resolve_field(
         return Ok(None);
     };
 
-    match CurrentResolverType::new(&field, ctx) {
+    let result = match CurrentResolverType::new(&field, ctx) {
         CurrentResolverType::PRIMITIVE => resolve_primitive_field(ctx, parent_type, field, parent_resolver_value).await,
         CurrentResolverType::CONTAINER => resolve_container_field(ctx, field, parent_resolver_value).await,
         CurrentResolverType::ARRAY => resolve_array_field(ctx, field, parent_resolver_value).await,
     }
-    .map(Some)
-    .map_err(|error| ctx.set_error_path(error))
+    .map_err(|error| ctx.set_error_path(error));
+
+    match result {
+        Ok(result) => Ok(Some(result)),
+        Err(e) if !field.ty.is_non_null() => {
```

Comment:
opt: `!field.ty.is_non_null()` is a bit hard to read, `field.ty.is_nullable()` would be simpler.